src/controllers/EventsController.js

```diff
@@ -1,176 +1,115 @@
 // @flow
 
-import moment from 'moment';
+import type { Event, EventData } from '../types';
+import type EventManager from '../managers/EventManager';
+
 import Controller from './Controller';
-import CoreController from '../lib/CoreController';
 import route from '../decorators/route';
 import httpVerb from '../decorators/httpVerb';
-
-import v1Api from '../views/api_v1';
+import serverSentEvents from '../decorators/serverSentEvents';
 import logger from '../lib/logger';
+import eventToApi from '../lib/eventToApi';
 
 class EventsController extends Controller {
-  _aliveInterval: ?string;
-  _lastMessage: ?Date;
-  _socket: ?Object;
+  _eventManager: EventManager;
 
-  _cleanup() {
-    try {
-      if (this._socket) {
-        this._socket.close();
-        this._socket = null;
-      }
-
-      if (this.response && this.response.socket) {
-        this.response.socket.end();
-        this.response.end();
-      }
-
-      if (this._aliveInterval) {
-        clearInterval(this._aliveInterval);
-        this._aliveInterval = null;
-      }
-    } catch (error) {
-      logger.error(`pipeEvents cleanup error: ${error}`);
-    }
+  constructor(eventManager: EventManager) {
+    super();
+
+    this._eventManager = eventManager;
   }
 
-  _keepAlive() {
-    if (((new Date()) - this._lastMessage) >= 9000) {
-      this._lastMessage = new Date();
-      this.response.write('\n');
-      this._checkSocket();
-    }
+  _closeStream(subscriptionID: string): Promise<void> {
+    return new Promise((resolve: () => void) => {
+      // TODO i'm not sure if we need all 4 listens here
+      this.request.on('close', () => {
```

Comment:
btw, I'm not sure but according to this: https://nodejs.org/api/http.html#http_event_close_1, may be I need to add reject() in 'close' case?